src/components/ShoppingListItem/ShoppingListItem.js

```diff
@@ -103,6 +105,11 @@ const ShoppingListItem = ({
     );
   }, [item]);
 
+  // change whether the item details are shown if showAllDetails is changed
+  useEffect(() => {
+    setShowSingleDetail(showAllDetails);
+  }, [showAllDetails]);
+
```

Comment:
This is great! We're changing the singleDetail state to showAllDetails if showAllDetails changes. Good learning opportunity for me. Thank you!